tests/kuery/test_lint.py

```diff
@@ -31,8 +31,8 @@ def test_upper_tokens(self):
         ]
 
         for q in queries:
-            with self.assertRaises(kql.KqlParseError):
-                kql.parse(q)
+            parsed_query = kql.parse(q)
+            self.assertNotEqual(str(parsed_query), q, f"Parsed query {parsed_query} matches the original {q}")
```

Comment:
can you update this to run twice - once as the original and once asserting no error is raised when normalization is True

Reply:
For some reason, I interpreted differently, but I could be mistaken so I have a couple questions:
1. Is this addition the second run? I saw Justin mentioned to run twice.
2. The second was suppose to NOT error so assertNotEqual sounds like it's doing something different. Any reason you went with this approach?

Reply:
1. The first run is the original check to see that the default behavior of using capital keywords results results in an error. The second check is to see if the the keyword case conversion functions correctly and makes all of the queries valid through changing them. 
2. The not equal is to ensure that the transform did actually change the original query. If it did not then something went wrong with  the token transformation/detection. I went with this approach originally to be able to have both tests combined into one. If the query parsing succeeds and it was not transformed, then it violates the first assumption that all of the queries should fail. That being said, I think having both tests makes it more readable what we are testing, but technically this second test makes first test redundant.